packages/cactus-plugin-ledger-connector-fabric/src/main/typescript/plugin-ledger-connector-fabric.ts

```diff
@@ -1063,6 +1114,70 @@ export class PluginLedgerConnectorFabric
+
+      // if we don't want to collect reads, than add condition  && transactionId !== ""
+      if (this.collectTransactionReceipts)  {
+        const startTimeFabricReceipt = new Date();
+        const txParams = req.params;
+        //getTransactionReceiptByTxID requires 2 params in req.params => channelName and txID
+        req.params = [];
+        req.params[0] = req.channelName;
+        req.params[1] = transactionId;
+        //req.params get are stored in the basicTxReceipt rwsetWriteData
+        if (transactionId)  {
+          const basicTxReceipt = await this.getTransactionReceiptByTxID(req);
+        const extendedReceipt: FabricV2TxReceipt={
+          caseID: req.caseID || "FABRIC_TBD",
+          transactionID: transactionId,
+          blockchainID: LedgerType.Fabric2,
+          invocationType: req.invocationType,
+          methodName: req.methodName,
+          parameters: txParams,
+          timestamp: new Date(),
+          channelName: req.channelName,
+          contractName: req.contractName,
+          signingCredentials: req.signingCredential,
+          endorsingParties: req.endorsingParties,
+          endorsingPeers: req.endorsingPeers,
+          gatewayOptions: req.gatewayOptions,
+          transactionCreator: basicTxReceipt.transactionCreator,
+          transientData: req.transientData,
+          blockMetaData: basicTxReceipt.blockMetaData,
+          chainCodeName: basicTxReceipt.chainCodeName,
+          blockNumber: basicTxReceipt.blockNumber,
+          chainCodeVersion: basicTxReceipt.chainCodeVersion,
+          responseStatus: basicTxReceipt.responseStatus,
+        };
+        const txReceipt = new amqp.Message(extendedReceipt);
+        this.amqpQueue?.send(txReceipt);
+        this.log.debug(`Sent extended transaction receipt to queue ${this.queueId}`);
+        } else  {
+          const extendedReceipt: FabricV2TxReceipt={
+            caseID: req.caseID || "FABRIC_TBD",
+            transactionID: undefined,
+            blockchainID: LedgerType.Fabric2,
+            invocationType: req.invocationType,
+            methodName: req.methodName,
+            parameters: txParams,
+            timestamp: new Date(),
+            channelName: req.channelName,
+            contractName: req.contractName,
+            signingCredentials: req.signingCredential,
+            endorsingParties: req.endorsingParties,
+            endorsingPeers: req.endorsingPeers,
+            gatewayOptions: req.gatewayOptions,
+            };
+          const txReceipt = new amqp.Message(extendedReceipt);
+          this.amqpQueue?.send(txReceipt);
+          this.log.debug(`Sent simple transaction receipt to queue ${this.queueId}`);
+        }
+        const endTimeFabricReceipt = new Date();
+        this.log.debug(`EVAL-${this.className}-GENERATE-AND-CAPTURE-RECEIPT:${endTimeFabricReceipt.getTime()-startTimeFabricReceipt.getTime()}`);
+
+      }
+
       const outUtf8 = out.toString("utf-8");
```

Comment:
@RafaelAPB I'd move all this logic to the cc viz plugin itself, have the connector plugin and the cc viz plugin decoupled from each other. If there must be some dependency from one to the other then I'd definitely go with the cc viz plugin depending on the connector plugin and not the other way around because the connector is a base plugin that provides something pretty much every application needs (connectivity to the ledger itself) while the cc viz plugin is a higher level feature.

Reply:
This means that a cctxviz plugin needs to receive connectors as input. How can I generate a modified transaction receipt? Would I have to implement a modified connector class? @petermetz

Reply:
@petermetz can we have input on this? I believe this is decisive for this PR to go onward (if not acceptable, we will just keep cctxviz as a fork from Cactus).

Reply:
> @petermetz can we have input on this? I believe this is decisive for this PR to go onward (if not acceptable, we will just keep cctxviz as a fork from Cactus).

@RafaelAPB Sorry again for the slow response! You can add the generic RxJS subject to the connectors that I spoke of and those will have a generic type parameter which can be the interface type that contains what is actually in your custom tx receipts (caseID for example).
Happy to provide code examples just let me know if this is not enough information!
The PR is definitely acceptable just needs some refactor to make it generic enough so that it doesn't hinder future improvements by hardcoding too many specifics in the connectors (such as AMQP)